### PR TITLE
componentDidUpdate with snapshot, updated debug macro, fixed typo in readme

### PR DESCRIPTION
**Pull request by Mansfield92**
…readme

### CODE REVIEW

File: src/lib/react/ReactDebugMacro.hx [hunk 1]
```diff
@@ -47,8 +47,13 @@ class ReactDebugMacro
 			{
 				switch (field.kind) {
 					case FFun(f):
+						#if react_snapshot_api
+						if (f.args.length != 2 && f.args.length != 3)
```
Comment (elsassph) [inline src/lib/react/ReactDebugMacro.hx]: Should be always 3, I think.

File: src/lib/react/ReactDebugMacro.hx [hunk 2]
```diff
@@ -85,10 +90,17 @@ class ReactDebugMacro
 					type: stateType,
 					opt: false,
 					value: null
+				},
+				{
```
Comment (elsassph) [inline src/lib/react/ReactDebugMacro.hx:95]: This should be conditional

File: test/src/react/ReactComponent.hx [hunk 3]
```diff
@@ -82,14 +82,25 @@ class ReactComponentOf<TProps, TState>
...
 	/**
 		https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html
 	**/
 	function componentDidCatch(error:Error, info:{ componentStack:String }):Void {}
 
+	/**
+		https://reactjs.org/docs/react-component.html#getsnapshotbeforeupdate
+		Note: this API has been introduced in React 16.3
+	**/
+	function getSnapshotBeforeUpdate(prevProps:TProps, prevState:TState):Dynamic;
```
Comment (elsassph) [inline test/src/react/ReactComponent.hx:103]: Should be conditional

File: test/src/react/ReactComponent.hx [hunk 4]
```diff
@@ -82,14 +82,25 @@ class ReactComponentOf<TProps, TState>
 
 	/**
 		https://facebook.github.io/react/docs/react-component.html#componentdidupdate
+		Note: Updated to version introduced in React 16.3
 	**/
-	function componentDidUpdate(prevProps:TProps, prevState:TState):Void {}
+	#if react_snapshot_api
+	function componentDidUpdate(prevProps:TProps, prevState:TState, ?snapshot:Dynamic):Void;
```
Comment (elsassph) [inline test/src/react/ReactComponent.hx]: Both shoud have an empty body `{}`

File: test/src/react/ReactComponent.hx (hunk 3 above)
Comment (elsassph) [reply test/src/react/ReactComponent.hx:103]: And should have an empty body.

File: src/lib/react/ReactDebugMacro.hx (hunk 1 above)
Comment (Mansfield92) [reply src/lib/react/ReactDebugMacro.hx]: Done

File: src/lib/react/ReactDebugMacro.hx (hunk 2 above)
Comment (Mansfield92) [reply src/lib/react/ReactDebugMacro.hx:95]: Done

File: test/src/react/ReactComponent.hx (hunk 3 above)
Comment (Mansfield92) [reply test/src/react/ReactComponent.hx:103]: Done

File: test/src/react/ReactComponent.hx (hunk 4 above)
Comment (Mansfield92) [reply test/src/react/ReactComponent.hx]: Done

**Pull request merged**